### PR TITLE
Refactor catalog, add API Card component

### LINKED ISSUE #176: Create API Card component
In preparation to work with the API Card, we should restructure the code a bit. The API Card markup currently embedded only in the API Catalog. Move the API Card markup out of the API Catalog template, into its own component.

### LINKED ISSUE #180: Refactor 'catalogue' component; rename to 'catalog'
While refactoring the 'API Card' component, it was apparent that the 'Catalogue' component needs refactoring. Refactor the catalogue component to be called 'API Catalog', simplify markup/classes, remove CSS where possible.

### PR DESCRIPTION
**Pull request by brylie**
Closes #176
Closes #180 
Related to #162

# Changes
- Reorganize templates
  - Rename API Catalog
  - Create API Card
- Cleanup
  - Remove unneeded styles, where possible
  - Use semantic markup, where possible
  - Semantic publicaiton/subscription name
  - Semantic template helpers
- Changes to routes
  - API Catalog (now '/apis')
  - Add API form (now '/apis/new')

# Known issues
- Template i18n tokens need to be updated to match new template names (after translation task for this sprint to avoid conflicts)

### CODE REVIEW

**Comment by marla-singer** [comment]
@brylie Reviewing

**Comment by marla-singer** [comment]
@brylie Change router for Sign out case. Not it redirects to 'catalogue' path

**Comment by marla-singer** [comment]
@brylie Also create separate issue for udated i18n tags and link with current issue

File: apis/client/latest/latest.js [hunk 1]
```diff
@@ -1,46 +1,25 @@
...
-    // Iterate through all documents
-    _.each(latestApiBackendsList, function (apiBackend) {
-
-      // Return to user human-readable timestamp
-      apiBackend.relative_created_at = moment(apiBackend.created_at).fromNow();
-    });
-
-    return latestApiBackendsList;
-  }
+    return Apis.find({}, { sort: { created_at: -1 } }).fetch();
```
Comment (marla-singer) [inline apis/client/latest/latest.js]: Delete sort parameter. It is redundant, you sort items on server side.

File: apis/client/latest/latest.html [hunk 2]
```diff
@@ -1,36 +1,14 @@
...
-        </div>
+<template name="latestPublicApis">
+  <h2 class="feature-section-heading text-center">
+    {{_ "latestApiBackends_Heading" }}
+  </h2>
+  <hr class="primary">
+  {{# if latestPublicApis }}
+    {{# each api in latestPublicApis }}
+      <div class="col-xl-2 col-lg-3 col-md-4 col-sm-6 col-xs-12">
+        {{> apiCard api=api }}
```
Comment (marla-singer) [inline apis/client/latest/latest.html]: Is it coordinate that API card is changed on front page?
**Was**
joxi_screenshot_1482491483805
**Now**
joxi_screenshot_1482491483805 1

Comment (marla-singer) [inline apis/client/latest/latest.html]: Not sure that we can change showcase on front page

File: api_catalog/client/toolbar/toolbar.html [hunk 3]
```diff
@@ -19,14 +18,14 @@
...
         <div class="nav navbar-nav">
-          <p class="navbar-text sort-filter-toolbar-title">
-            {{_ "catalogue_toolbar_sortBy_title" }}
-          </p>
-          <div class="form-group sort-options-form">
-            <form class="form navbar-form navbar-nav" id="sort-options-select">
-              <select name="sort-menu" class="form-control sort-options-select" id="sort-select">
+          <form class="navbar-form">
+            <div class="form-group">
+              <label for="sort-menu">
```
Comment (marla-singer) [inline api_catalog/client/toolbar/toolbar.html]: Use value of attribute **for** the same as id of releted element. Now no one element with id "sort-menu"

File: apis/client/latest/latest.js (hunk 1 above)
Comment (brylie) [inline apis/client/latest/latest.js]: Sort needs to be done on both client and server:

http://meteor.hromnik.com/blog/how-to-sort-collections-in-meteor

File: apis/client/latest/latest.html (hunk 2 above)
Comment (brylie) [inline apis/client/latest/latest.html]: We are trying to re-use the API Card component, for consistency. This means we can replace the redundant code on the front page.

File: api_catalog/client/toolbar/toolbar.html (hunk 3 above)
Comment (brylie) [inline api_catalog/client/toolbar/toolbar.html]: OK, I fixed the label for.

File: apis/client/delete/delete.js [hunk 4]
```diff
@@ -6,12 +6,15 @@ import { sAlert } from 'meteor/juliancwirko:s-alert';
 import { Modal } from 'meteor/peppelg:bootstrap-3-modal';
 
 Template.deleteApiConfirmation.events({
-  'click #modal-delete-api': function (event, instance) {
+  'click #modal-delete-api': function () {
+    // Get reference to template instance
+    const instance = Template.instance();
+
```
Comment (marla-singer) [inline apis/client/delete/delete.js]: Why do you delete args and initialize Template instance?

Comment (brylie) [inline apis/client/delete/delete.js]: Ah, right. I was linting.

Comment (brylie) [inline apis/client/delete/delete.js]: Fixed. Was confused by lint messages, I think.

**Comment by marla-singer** [comment]
@brylie Merge it

**Pull request merged**